server/fleet/microsoft_mdm.go

```diff
@@ -199,14 +199,49 @@ func (req *SoapRequest) IsValidRequestSecurityTokenMsg() error {
 		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.ValueType")
 	}
 
-	if len(req.Body.RequestSecurityToken.BinarySecurityToken.EncodingType) == 0 {
-		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.EncodingType")
+	if req.Body.RequestSecurityToken.BinarySecurityToken.ValueType != mdm.EnrollReqTypePKCS10 &&
+		req.Body.RequestSecurityToken.BinarySecurityToken.ValueType != mdm.EnrollReqTypePKCS7 {
+		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.EncodingType  not supported")
 	}
 
 	if len(req.Body.RequestSecurityToken.BinarySecurityToken.Content) == 0 {
 		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.Content")
 	}
 
+	if len(req.Body.RequestSecurityToken.AdditionalContext.ContextItems) == 0 {
+		return errors.New("invalid requestsecuritytoken message: AdditionalContext.ContextItems")
```

Comment:
It fails if the len of context items is zero, so I think the message should say something like "AdditionalContext.ContextItems missing"?

Reply:
agreed, just added this